Assets/Scripts/Game/LevitateMotor.cs

```diff
@@ -68,17 +68,25 @@ private void Update()
             if (GameManager.Instance.PlayerEntity.IsParalyzed)
                 return;
 
+            float inputX = 0f;
```

Comment:
Hey, sorry for the wait! I have been traveling and I finally got a chance to take a look at this.

After looking at the FrictionMotor, what it does is simply set the inputX and inputY to
```
float inputX = InputManager.Instance.Horizontal;
float inputY = InputManager.Instance.Vertical;
```
I tested the code using this instead of that chunk of code dealing with HasAction() and it looks like it works perfectly, so I think you can remove that entire chunk and use that instead.

Reply:
Thanks for the feedback!
I have a concern though, InputManager.Instance.Horizontal is not bound by -1 .. -1 but by -NegHorizontalLimit .. PosHorizontalLimit, etc. so shouldn't they be scaled before use, something like
```
InputManager IM = InputManager.Instance;
float inputX = (IM.Horizontal + IM.NegHorizontalLimit) / (IM.PosHorizontalLimit + IM.NegHorizontalLimit);
float inputY = (IM.Vertical + IM.NegVerticalLimit) / (IM.PosVerticalLimit + IM.NegVerticalLimit);
```

Reply:
I might be wrong but I think it may have been intended to limit the top speed of the player, so we might not need to have scaling. @ajrb wrote the limits so I think he can answer that.

Reply:
The ability to change the limit of maximum movement in each direction was added to enable different speeds in various directions by mods. It's used in the enhanced riding module of R&R mod to limit the speed you can move sideways or backwards when riding a horse. It always seemed odd to be able to move backwards as fast as you could forward on a horse.

I think when levitation is involved, in my use case of the limits at least, there's no reason to limit movement in any direction. I think it would actually be bad for the normal grounded movement limits to also affect magical flying.

Reply:
Shouldn't it be up to the mods to decide what limits should be applied during walk/swimming/riding/levitation/whatever?
Are you okay with with PR as-is?

Reply:
I think it looks good!

Reply:
I did what I needed for my mod, and havn't see any calls for a more complex system. I am happy with the PR as-is w/o any levitation limiting.